### DISCUSSION TITLE
Hold shift to extend from the last drawn point

### DISCUSSION THREAD

**Discussion by kendofriendo** [Orama-Interactive/Pixelorama#1129] [Ideas]
In Aseprite and Photoshop, if you draw a pixel on the canvas, let go of the button and move the mouse, pressing shift will show a line extending from that first pixel to the cursor. If you click again while holding shift a line will be drawn between the two points. This is my most frequently used technique for drawing with a mouse, and as far as I can tell it's not possible in Pixelorama.

**Comment by Variable-ind** [comment]
Hy, this should be possible with the curve tool i think (it can draw straight lines too)

**Reply by kendofriendo** [reply]
The curve tool only creates an outline, then draws when two points connect. It's also much more difficult to control as you have to either double click to draw each line or connect to the last point. It also "tries" to curve by rendering two or more pixels at a time. The shift-to-connect feature gives you pixel-by-pixel control, letting you draw complex shapes on the fly.

**Reply by Variable-ind** [reply]
> The curve tool only creates an outline, then draws when two points connect. It's also much more difficult to control as you have to either double click to draw each line or connect to the last point. It also "tries" to curve by rendering two or more pixels at a time. The shift-to-connect feature gives you pixel-by-pixel control, letting you draw complex shapes on the fly.

(I am NOT against the feature, infact i like it)

(Edit: it seems you already know about this one too) Btw, curve tool can draw without connecting two end points as well (click, move somewhere then double click).
Click adds a point,
Shift removes last added point,
Click + drag, curves the line,
Double click applies the preview.
